controllers/activemqartemis_reconciler.go

```diff
@@ -2756,6 +2761,47 @@ func getPropertiesResourceNsName(artemis *brokerv1beta1.ActiveMQArtemis) types.N
 	}
 }
 
+// applyControlPlaneOverrides applies control plane configuration overrides from secrets.
+// It first checks for CR-specific override secret ([cr-name]-control-plane-override),
+// then falls back to shared override secret (control-plane-override).
+// Each key in the override secret completely replaces the corresponding key in brokerPropertiesMapData.
+func applyControlPlaneOverrides(customResource *brokerv1beta1.ActiveMQArtemis, client rtclient.Client, brokerPropertiesMapData map[string]string) error {
+	ctx := context.Background()
+
+	// Try CR-specific override secret first
```

Comment:
one small observation. we have logic that deals with cr specific and generic secrets in a few places, is there an opportunity to combine those into some function that we can make use of here. 
I note that in some cases we want to error if not found, unlike this case, but that can be in a method name, resolveSecret, resolveRequiredSecret etc